### PR TITLE
test sha256 injection

### PR DESCRIPTION
**Pull request by ipa-mdl**
tests https://github.com/osrf/docker_templates/pull/90

### CODE REVIEW

File: gazebo/10/ubuntu/bionic/libgazebo10/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # This is an auto generated Dockerfile for gazebo:libgazebo10
 # generated from docker_images/create_gzclient_image.Dockerfile.em
 FROM gazebo:gzserver10-bionic
+# label gazebo packages
```
Comment (ruffsl) [inline gazebo/10/ubuntu/bionic/libgazebo10/Dockerfile:4]: Nit pick: could you tweak the templates so there is always a space before line comments.

File: ros/dashing/ubuntu/bionic/ros1-bridge/Dockerfile [hunk 2]
```diff
@@ -10,13 +10,23 @@ RUN echo "deb http://packages.ros.org/ros/ubuntu bionic main" > /etc/apt/sources
 
 ENV ROS1_DISTRO melodic
 ENV ROS2_DISTRO dashing
+# label ros packages
```
Comment (ruffsl) [inline ros/dashing/ubuntu/bionic/ros1-bridge/Dockerfile:13]: Same new line nitpick.

File: ros/lunar/debian/stretch/desktop-full/Dockerfile [hunk 3]
```diff
@@ -2,8 +2,11 @@
 # generated from docker_images/create_ros_image.Dockerfile.em
 FROM osrf/ros:lunar-desktop-stretch
 
+# label ros packages
```
Comment (ruffsl) [inline ros/lunar/debian/stretch/desktop-full/Dockerfile:5]: Not sure if we should change EOL Dockerfile. I not against it, as I like consistency, but we should check for side effects.

File: ros/lunar/debian/stretch/ros-core/Dockerfile [hunk 4]
```diff
@@ -14,24 +14,17 @@ RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4B63CF8FDE
...
-    && rm -rf /var/lib/apt/lists/*
-
 # setup environment
 ENV LANG C.UTF-8
 ENV LC_ALL C.UTF-8
 
-# bootstrap rosdep
-RUN rosdep init \
-    && rosdep update --include-eol-distros
+ENV ROS_DISTRO lunar
```
Comment (ruffsl) [inline ros/lunar/debian/stretch/ros-core/Dockerfile:21]: Not sure if we should change EOL Dockerfile. I not against it, as I like consistency, but we should check for side effects.

File: ros/lunar/debian/stretch/ros-core/Dockerfile [hunk 5]
```diff
@@ -14,24 +14,17 @@ RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4B63CF8FDE
...
+
+# label ros packages
+LABEL sha256.ros-lunar-ros-core=cdf4fa8a555d13e8d69b7bf2fd69d8169d36949d40ee7700a642b9a125f77204
```
Comment (ruffsl) [inline ros/lunar/debian/stretch/ros-core/Dockerfile:24]: Something @mikaelarguedas mentioned to me. Is this the same sha the is in the snapshot repo?

**Comment by mikaelarguedas** [comment]
> Not sure if we should change EOL Dockerfile. I not against it, as I like consistency, but we should check for side effects.

:+1: I like the ideas that the images not rebuilt by dockerhub anymore dont see changes. The fact that only non-EOL (active in the manifest file) Dockerfile get updated sits well with me.

An easy way to just regenerate those is to use the global gneerate_dockerfolders.py script:
```
root@5f28475d0a7e:/tmp/docker_images/ros# ./create_dockerfolders.py dir -d . -a
root@5f28475d0a7e:/tmp/docker_images/gazebo# ./create_dockerfolders.py dir -d . -a
```

File: ros/lunar/debian/stretch/ros-core/Dockerfile (hunk 5 above)
Comment (mikaelarguedas) [reply ros/lunar/debian/stretch/ros-core/Dockerfile:24]: If we decide to not update EOL images (suggested in https://github.com/osrf/docker_images/pull/454#issuecomment-689149674) we should end up with all the Dockerfiles using snapshots.ros.org remaining unchanged in this PR.

If we decide to update them anyways, we should check that the sha is matching the one from snapshots.ros.org.

**Comment by mikaelarguedas** [comment]
Unfortunately some Dockerfiles have diverged from the templates for the time being.
Some of the changes we'll need to keep as is and not override:
- EOL images (the ones not generated by the commands in https://github.com/osrf/docker_images/pull/454#issuecomment-689149674)
- installation of the ros bootstrap tools (in noetic) should be kept as `python3-<package_name>`

**Review by mikaelarguedas — COMMENTED** [review]
This is really great thanks @ipa-mdl for working on this. This looks good and aside from a couple nitpicks it's almost ready for prime time 

I made a couple comments about the files that IMO should not be modified to stay true to their last image built on dockerhub as well as some "not matching the template" installs in noetic that we should keep for the resulting image to stay usable

**Comment by ruffsl** [comment]
Closing for https://github.com/osrf/docker_images/pull/457

**Pull request closed without merging**